c/src/ml-api-common.c

```diff
@@ -714,8 +714,7 @@ _ml_tensors_data_destroy_internal (ml_tensors_data_h data, gboolean free_data)
     } else {
       for (i = 0; i < ML_TENSOR_SIZE_LIMIT; i++) {
         if (_data->tensors[i].data) {
-          g_free (_data->tensors[i].data);
-          _data->tensors[i].data = NULL;
+          g_clear_pointer (&_data->tensors[i].data, g_free);
```

Comment:
unnecessary condition `if (_data->tensors[i].data)', g_clear_pointer() internally checks ptr.

Reply:
You're right. Thank you.